### ISSUE TITLE
[Privileges 2] Randomly sended webhooks "privileges expired"

### ISSUE THREAD

**Issue by aduffner** [SAP/macOS-enterprise-privileges#124]
Tested on:
- macOS 15.0.1 (24A348)
- Privileges Beta 2.0.0 (1815)
<img width="1590" alt="Bildschirmfoto 2024-10-28 um 11 02 19" src="https://github.com/user-attachments/assets/2854d91e-b9b9-4400-8510-68d0dfddb4d6">

**Comment by mthielemann** [comment]
@aduffner Weird. Could you please check if the agent or daemon crashed? Are there any crash reports in `/Library/Logs/Diagnostic Reports`?

**Comment by aduffner** [comment]
@mthielemann  I can't see anything relevant, which file might be interesting?
```
$ sudo ls "/Library/Logs/DiagnosticReports"
.Microsoft Outlook_2023-07-02-002945_<COMPNAME>.diag
.WindowServer_2023-04-12-180348_<COMPNAME>.wakeups_resource.diag
.WindowServer_2024-01-23-011127_<COMPNAME>.wakeups_resource.diag
.com.apple.MobileSoftwareUpdate.UpdateBrainService_2024-03-26-111522_<COMPNAME>.diag
.contents.panic
Cyberduck_2024-10-21-133210_<COMPNAME>.hang
Google Chrome Helper_2024-10-24-153705_<COMPNAME>.diag
Google Chrome Helper_2024-10-25-154601_<COMPNAME>.diag
JetsamEvent-2024-10-26-064946.ips
Microsoft Outlook_2024-10-22-152835_<COMPNAME>.diag
Preview_2024-10-22-140231_<COMPNAME>.hang
ProxiedDevice-251c419f8becbe5ae13825a41b85ef6f7bd5085d
ProxiedDevice-ea71178b2ea4c86172eab6ee380641de5288aab1
Retired
ditto_2024-10-25-100129_<COMPNAME>.diag
duetexpertd_2024-10-21-165947_<COMPNAME>.cpu_resource.diag
duetexpertd_2024-10-24-061806_<COMPNAME>.cpu_resource.diag
duetexpertd_2024-10-26-221005_<COMPNAME>.cpu_resource.diag
io.phorion.tcc-kronos.TCCKronosExtension_2024-10-26-012254_<COMPNAME>.diag
knowledgeconstructiond_2024-10-21-173626_<COMPNAME>.cpu_resource.diag
knowledgeconstructiond_2024-10-23-190746_<COMPNAME>.cpu_resource.diag
knowledgeconstructiond_2024-10-25-122750_<COMPNAME>.cpu_resource.diag
logd_2024-10-26-043135_<COMPNAME>.diag
shutdown_stall_2024-10-24-183347_<COMPNAME>.shutdownStall
signpost_reporter_2024-10-26-112021_<COMPNAME>.cpu_resource.diag
spotlightknowledged_2024-10-21-181255_<COMPNAME>.cpu_resource.diag
spotlightknowledged_2024-10-22-123316_<COMPNAME>.cpu_resource.diag
spotlightknowledged_2024-10-24-125001_<COMPNAME>.cpu_resource.diag
spotlightknowledged_2024-10-25-121320_<COMPNAME>.cpu_resource.diag
spotlightknowledged_2024-10-25-163023_<COMPNAME>.cpu_resource.diag
spotlightknowledged_2024-10-26-075450_<COMPNAME>.cpu_resource.diag
spotlightknowledged_2024-10-26-114433_<COMPNAME>.cpu_resource.diag
spotlightknowledged_2024-10-27-055625_<COMPNAME>.cpu_resource.diag
spotlightknowledged_2024-10-27-062213_<COMPNAME>.cpu_resource.diag
spotlightknowledged_2024-10-28-040003_<COMPNAME>.cpu_resource.diag
spotlightknowledged_2024-10-28-044649_<COMPNAME>.cpu_resource.diag
storagekitd_2024-10-22-103730_<COMPNAME>.diag
suggestd_2024-10-25-160807_<COMPNAME>.cpu_resource.diag
telemetryd_v2_2024-10-24-104138_<COMPNAME>.diag
```

**Comment by mthielemann** [comment]
And in `/Library/Logs/Crash Reporter`?

**Comment by mthielemann** [comment]
Is there anything I can do to reproduce it on my end?

**Comment by aduffner** [comment]
I am not sure, here you have maybe a more complete screenshot:
<img width="1559" alt="Bildschirmfoto 2024-10-28 um 11 42 21" src="https://github.com/user-attachments/assets/94d616b3-4188-43b6-bc18-cf3df387e9ba">

What's interesting:
The difference between the "false" time stamps is always 30, 60, 120, 180, 240 or 300 seconds.
Sometimes every minute, sometimes first after 5 minutes.

Currently I am using following config:
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
  <dict>
    <key>PostChangeExecutablePath</key>
    <string></string>
    <key>PostChangeActionOnGrantOnly</key>
    <true/>
    <key>ExpirationIntervalMax</key>
    <integer>240</integer>
    <key>RequireAuthentication</key>
    <true/>
    <key>AllowCLIBiometricAuthentication</key>
    <true/>
    <key>RevokePrivilegesAtLogin</key>
    <true/>
    <key>RemoteLogging</key>
    <dict>
      <key>ServerType</key>
      <string>webhook</string>
      <key>ServerAddress</key>
      <string>myURL</string>
    </dict>
  </dict>
</plist>
```

**Comment by mthielemann** [comment]
@aduffner Just sent you a PM.

**Comment by mthielemann** [comment]
Tested with your configuration but was unable to reproduce the issue. Just wondering about all the "Failed to connect to daemon" messages you are seeing. I don't see this on my end. 🤷🏻‍♂️

**Comment by aduffner** [comment]
@mthielemann Very good that you were able to verify that “Failed to connect to daemon” does not occur for you. That got me thinking and I found files of Privileges Demoter on the system, which I removed completely via: https://github.com/sgmills/PrivilegesDemoter/blob/main/Uninstall%20PrivilegesDemoter.sh

Now the error message no longer occurs and I hope that the behavior will also disappear. I have not yet fully understood how the interaction comes about, but I strongly suspect that it was due to this.
